docs/gcp_setup.md

```diff
@@ -0,0 +1,287 @@
+storage.objects.get,storage.objects.list \
+--stage=GA
+
+gcloud storage buckets add-iam-policy-binding gs://${BUCKET} \
+--member=serviceAccount:${SVC_ACCOUNT}@${PROJECT}.iam.gserviceaccount.com \
+--role=projects/${PROJECT}/roles/${ROLE}
+```
+
+#### goe_bq_core_role
+Note that this role is granted at the project level.
+```
+gcloud iam roles create goe_bq_core_role --project ${PROJECT} \
+--title="GOE Core BigQuery Access" \
+--description="GOE permissions for core access to BigQuery" \
+--permissions=bigquery.datasets.get,\
+bigquery.jobs.create,\
+bigquery.tables.get,\
+bigquery.tables.list \
+--stage=GA
+
+gcloud projects add-iam-policy-binding ${PROJECT} \
+--member=serviceAccount:${SVC_ACCOUNT}@${PROJECT}.iam.gserviceaccount.com \
+--role=projects/${PROJECT}/roles/goe_bq_core_role
+```
+
+If GOE is permitted to create datasets then add these extra privileges:
+```
+gcloud iam roles update goe_bq_core_role --project ${PROJECT} \
+--add-permissions=bigquery.datasets.create
+```
+
+#### goe_bq_app_role
+Note that the role grant is bound to the target BigQuery dataset, no project wide access is granted. The `bq` utility is used to grant the role because `gcloud` does not support these granular grants.
+
+Also note that the target dataset must be created *before* executing these commands (see [Generating Dataset DDL](#generating-dataset-ddl) for details).
+```
+gcloud iam roles create goe_bq_app_role --project ${PROJECT} \
+--title="GOE Data Update Access" \
+--description="Grants GOE permissions to read and modify data in an application schema" \
+--permissions=bigquery.tables.getData,\
+bigquery.tables.updateData \
+--stage=GA
+
+echo "GRANT \`projects/${PROJECT}/roles/goe_bq_app_role\` ON SCHEMA ${TARGET_DATASET}
+TO \"serviceAccount:${SVC_ACCOUNT}@${PROJECT}.iam.gserviceaccount.com\";
+" | bq query --project_id=${PROJECT} --nouse_legacy_sql --location=${LOCATION}
+```
+
+If GOE is permitted to create tables then add these extra privileges:
+```
+gcloud iam roles update goe_bq_app_role --project ${PROJECT} \
+--add-permissions=bigquery.tables.create,bigquery.tables.update
+```
+
+If GOE is permitted to drop tables then add these extra privileges:
+```
+gcloud iam roles update goe_bq_app_role --project ${PROJECT} \
+--add-permissions=bigquery.tables.delete
+```
+
+#### goe_bq_stg_role
+Note that the role grant is bound to the staging BigQuery dataset (which has the same name as the target dataset but with an "_load" suffix), no project wide access is granted. The `bq` utility is used to grant the role because `gcloud` does not support these granular grants.
+
+Also note that the staging dataset must be created *before* executing these commands (see [Generating Dataset DDL](#generating-dataset-ddl) for details).
+```
+gcloud iam roles create goe_bq_stg_role --project ${PROJECT} \
+--title="GOE Data Staging Access" \
+--description="Grants GOE permissions to manage objects in a staging schema" \
```

Comment:
Do we want to say "dataset" instead of "schema"?

Reply:
Fixed